File: src/main/CodeInjection.scala
import javax.script._

// CWE-94
object CodeInjection {
  def main(args: Array[String]): Unit = {
    try {
      val manager = new ScriptEngineManager
      val engine = manager.getEngineByName("JavaScript")
      System.out.println(args(0))
      engine.eval("print('" + args(0) + "')")
    } catch {
      case e: Exception =>
        e.printStackTrace()
    }
  }
}

// CWE-94
object CodeInjectionTestCx {
  def main(args: Array[String]): Unit = {
    try {
      val manager = new ScriptEngineManager
      val engine = manager.getEngineByName("JavaScript")
      System.out.println(args(0))
      engine.eval("print('" + args(0) + "')")
    } catch {
      case e: Exception =>
        e.printStackTrace()
    }
  }
}

File: src/main/CodeInjection.java
import javax.script.*;

// CWE-94
public class CodeInjection {
    public static void main(String[] args) {
        try {
            ScriptEngineManager manager = new ScriptEngineManager();
            ScriptEngine engine = manager.getEngineByName("JavaScript");
            System.out.println(args[0]);
            engine.eval("print('"+ args[0] + "')");
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
}